### PR TITLE
dtl: Support Basic Authentication for RPC endpoints

### PR DESCRIPTION
**Pull request by Inphi**
**Description**
Adds optional support for [Basic Authentication ](https://en.wikipedia.org/wiki/Basic_access_authentication) for RPC endpoints. 

Note: This is technically a breaking change because prior to this patch we inadvertently support multiple RPC endpoints as a csv, via the FallbackProvider. Now only a single endpoint will work.

**Metadata**
ENG-2062

### CODE REVIEW

**Comment by tynes** [comment]
We previously only did minor releases when we did a regenesis. I'm not super opinionated on whether or not we should continue not following semver the way that it was supposed to be used

**Comment by tynes** [comment]
We previously only did minor releases when we did a regenesis. I'm not super opinionated on whether or not we should continue not following semver the way that it was supposed to be used

File: packages/data-transport-layer/src/services/l1-ingestion/service.ts [hunk 1]
```diff
@@ -114,8 +110,11 @@ export class L1IngestionService extends BaseService<L1IngestionServiceOptions> {
     this.l1IngestionMetrics = registerMetrics(this.metrics)
 
     if (typeof this.options.l1RpcProvider === 'string') {
-      this.state.l1RpcProvider = FallbackProvider(this.options.l1RpcProvider, {
-        'User-Agent': 'data-transport-layer',
+      this.state.l1RpcProvider = new StaticJsonRpcProvider({
```
Comment (tynes) [inline packages/data-transport-layer/src/services/l1-ingestion/service.ts:113]: `proxyd` should be able to handle the retry logic now, so making this change should be fine

Comment (tynes) [inline packages/data-transport-layer/src/services/l1-ingestion/service.ts:113]: `proxyd` should be able to handle the retry logic now, so making this change should be fine

**Comment by mslipper** [comment]
What's the context for adding auth? Is there an upstream provider that requires this?

**Comment by Inphi** [comment]
> We previously only did minor releases when we did a regenesis. I'm not super opinionated on whether or not we should continue not following semver the way that it was supposed to be used

I see. I can make it patch in that case. We should have a discussion later about this though.

**Review by mslipper — APPROVED** [review]

**Comment by tynes** [comment]
`proxyd` supports auth, so this would allow the dtl to consume `proxyd` that is set up with auth

**Review by tynes — APPROVED** [review]

**Pull request merged**